wandb/sdk/artifacts/storage_policies/wandb_storage_policy.py

```diff
@@ -398,5 +402,9 @@ def _write_cache(self, entry: "ArtifactManifestEntry") -> None:
             try:
                 with cache_open("wb") as f, open(entry.local_path, "rb") as src:
                     shutil.copyfileobj(src, f)
+                if entry.local_path.startswith(staging_dir):
+                    # Delete staged files as soon as they're copied to the cache
+                    # instead of waiting till all the files are uploaded
+                    os.remove(entry.local_path)
```

Comment:
This is a theoretically isolated change from the rest of this PR, right? Even though it's small, I would like to put it in a separate PR because it seems like the kind of subtle change that could cause an unexpected regression. (non specifically, I don't have a particular bug in mind)